### PR TITLE
lighthouse: init module

### PR DESCRIPTION
**Pull request by happysalada**
###### Description of changes

mostly inspired from https://github.com/NixOS/nixpkgs/pull/128984/files
with a couple of tweaks.
@centromere @bachp you might be interested.

currently testing still.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/blockchain/ethereum/lighthouse.nix [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        example = "";
+      };
+    };
+  };
+
+  config = mkIf (cfg.beacon.enable || cfg.validator.enable) {
+
+    environment.systemPackages = [ pkgs.lighthouse ] ;
+
+    networking.firewall = mkIf cfg.beacon.enable {
```
Comment (bachp) [inline nixos/modules/services/blockchain/ethereum/lighthouse.nix:198]: We should not automatically open ports in a service. See https://github.com/NixOS/nixpkgs/issues/19504#issuecomment-253458728

**Review by bachp — CHANGES_REQUESTED** [review]

File: nixos/modules/services/blockchain/ethereum/lighthouse.nix (hunk 1 above)
Comment (happysalada) [reply nixos/modules/services/blockchain/ethereum/lighthouse.nix:206]: added an option for it that defaults to false.

**Comment by happysalada** [comment]
I'm going to merge this as I think it's usable and a better module will come from more feedback.
I would be happy to address any feedback in subsequent MRs.

**Comment by emmanuelrosa** [comment]
I don't use this module, yet as of f4ff9c26499a600fcc2459cfc6d8bbc5c1dfc5ea ...

```
nixos-rebuild build --flake .
building the system configuration...
error: The option `services.lighthouse.beacon' is used but not defined.
(use '--show-trace' to show detailed location information)
```

**Comment by bobby285271** [comment]
Yeah I also ran into similar issue when running a cinnamon test on ofborg several minutes ago:

```
error: The option `nodes.machine.services.lighthouse.beacon' is used but not defined.
```

https://github.com/NixOS/nixpkgs/runs/9335720683
https://logs.nix.ci/?attempt_id=84b5a0bb-6b0e-423a-ac40-1050cdae6146&key=nixos%2Fnixpkgs.200018

**Comment by happysalada** [comment]
sorry about that https://github.com/NixOS/nixpkgs/pull/200025

**Pull request merged**